rest/config_startup.go

```diff
@@ -85,6 +85,8 @@ type StartupConfig struct {
 	CouchbaseKeepaliveInterval *int   `json:"couchbase_keepalive_interval,omitempty" help:"TCP keep-alive interval between SG and Couchbase server"`
 
 	DeprecatedConfig *DeprecatedConfig `json:"-,omitempty" help:"Deprecated options that can be set from a legacy config upgrade, but cannot be set from a 3.0 config."`
+
+	avoidLoggingSetup bool `json:"-"` // Used to avoid logging setup so as to not modify globals. This only used for testing multiple ServerContexts simultaneously. This will use the same logging configuration as is already set up.
```

Comment:
Do you think we could build this into initLogging, so that callers don't need to worry about setting this flag?  (i.e. have InitLogging run within a mutex and set a global flag when complete, and check that flag when run)

Reply:
There's a single test `TestConfigsIncludeDefaults` that requires initialization of a logger, to test whether it will pass the `DCP` base key to the database level. I'm not convinced this part of this test needs the workaround `defer RunBootstrapLoggerInitialization(t)()`.

I don't know if I like this solution better because it relies on a global but I think it makes the part of the code to be modified very small, which I like.

Reply:
I needed to add `config.Logging.Console.Enabled = base.BoolPtr(true)` to this test, which is normally enabled as a side effect of `InitLogging` but not normally needed in a JSON for server config. I think this is fine for this test, which keeps the guts of the inheritance.